### PR TITLE
kyber v2 audit fixes

### PR DESCRIPTION
**Pull request by ilanDoron**

### CODE REVIEW

File: contracts/FeeBurner.sol [hunk 1]
```diff
@@ -98,37 +97,37 @@ contract FeeBurner is Withdrawable, FeeBurnerInterface, Utils {
 
     // this function is callable by anyone
     function burnReserveFees(address reserve) public {
-        uint burnAmount = reserveFeeToBurn[reserve];
+        uint burnAmount = reserveFeeToBurn[reserve] - 1;
```
Comment (yaronvel) [inline contracts/FeeBurner.sol]: this is bad right? might have an overflow, why did you change it?

File: contracts/FeeBurner.sol [hunk 2]
```diff
@@ -79,12 +78,12 @@ contract FeeBurner is Withdrawable, FeeBurnerInterface, Utils {
         require(fee >= walletFee);
         uint feeToBurn = fee - walletFee;
 
-        if (walletFee > 0) {
+        if (walletFee != 0) {
```
Comment (yaronvel) [inline contracts/FeeBurner.sol]: don't like this optimization

File: contracts/FeeBurner.sol [hunk 3]
```diff
@@ -79,12 +78,12 @@ contract FeeBurner is Withdrawable, FeeBurnerInterface, Utils {
...
             reserveFeeToWallet[reserve][wallet] += walletFee;
             AssignFeeToWallet(reserve, wallet, walletFee);
         }
 
-        if (feeToBurn > 0) {
+        if (feeToBurn != 0) {
```
Comment (yaronvel) [inline contracts/FeeBurner.sol]: don't like it

File: contracts/FeeBurner.sol [hunk 4]
```diff
@@ -98,37 +97,37 @@ contract FeeBurner is Withdrawable, FeeBurnerInterface, Utils {
...
+        require(burnAmount > 1);
         reserveFeeToBurn[reserve] = 1; // leave 1 twei to avoid spikes in gas fee
         if (taxWallet != address(0) && taxFeeBps != 0) {
-            taxToSend = (burnAmount - 1) * taxFeeBps / 10000;
-            require(burnAmount - 1 > taxToSend);
+            taxToSend = (burnAmount) * taxFeeBps / 10000;
+            require(burnAmount > taxToSend);
             burnAmount -= taxToSend;
-            if (taxToSend > 0) {
+            if (taxToSend != 0) {
```
Comment (yaronvel) [inline contracts/FeeBurner.sol]: hate this != instead of >. please revert all of them

File: contracts/FeeBurner.sol [hunk 5]
```diff
@@ -98,37 +97,37 @@ contract FeeBurner is Withdrawable, FeeBurnerInterface, Utils {
...
-        BurnAssignedFees(reserve, msg.sender, (burnAmount - 1));
+        BurnAssignedFees(reserve, msg.sender, burnAmount);
     }
 
     event SendWalletFees(address indexed wallet, address reserve, address sender);
 
     // this function is callable by anyone
     function sendFeeToWallet(address wallet, address reserve) public {
-        uint feeAmount = reserveFeeToWallet[reserve][wallet];
-        require(feeAmount > 1);
```
Comment (yaronvel) [inline contracts/FeeBurner.sol]: changes here seem too risky

File: contracts/KyberNetwork.sol [hunk 6]
```diff
@@ -182,6 +183,7 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
             require(whiteListContract != address(0));
             require(feeBurnerContract != address(0));
             require(expectedRateContract != address(0));
+            require(kyberNetworkProxyContract != address(0));
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: did you test it?

File: contracts/KyberNetwork.sol [hunk 7]
```diff
@@ -263,24 +265,25 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
...
         uint bestRate = 0;
         uint bestReserve = 0;
         uint numRelevantReserves = 0;
+        address[] memory reserveArr;
 
         //return 1 for ether to ether
-        if (src == dest) return (reserves[bestReserve], PRECISION);
-
-        address[] memory reserveArr;
+        if (src == dest) { return (reserves[bestReserve], PRECISION); }
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: why did you change it? why did you make changes in this function?

File: contracts/KyberNetwork.sol [hunk 8]
```diff
@@ -295,7 +298,7 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
             }
         }
 
-        if (bestRate > 0) {
+        if (bestRate != 0) {
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: ugly please revert all of these

File: contracts/KyberNetwork.sol [hunk 9]
```diff
@@ -356,31 +359,27 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
...
     }
 
-    /* solhint-disable function-max-lines */
     /// @notice use token address ETH_TOKEN_ADDRESS for ether
     /// @dev trade api for kyber network.
     /// @param tradeInput structure of trade inputs
     function trade(TradeInput tradeInput) internal returns(uint) {
         require(isEnabled);
         require(tx.gasprice <= maxGasPriceValue);
-        require(validateTradeInput(tradeInput.src, tradeInput.srcAmount, tradeInput.destAddress));
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: should still use require

File: contracts/KyberNetwork.sol [hunk 10]
```diff
@@ -356,31 +359,27 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
...
+        validateTradeInput(tradeInput.src, tradeInput.srcAmount, tradeInput.dest, tradeInput.destAddress);
 
         BestRateResult memory rateResult =
-        findBestRateTokenToToken(tradeInput.src, tradeInput.dest, tradeInput.srcAmount);
+            findBestRateTokenToToken(tradeInput.src, tradeInput.dest, tradeInput.srcAmount);
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: what is this new space?

File: contracts/KyberNetwork.sol [hunk 11]
```diff
@@ -356,31 +359,27 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
...
 
-        require(rateResult.rate > 0);
+        require(rateResult.rate != 0);
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: hate it

File: contracts/KyberNetwork.sol [hunk 12]
```diff
@@ -392,54 +391,41 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
         }
 
         // verify trade size is smaller than user cap
-        require(ethAmount <= getUserCapInWei(tradeInput.trader));
+        require(weiAmount <= getUserCapInWei(tradeInput.trader));
 
         //do the trade
         //src to ETH
-        require(doReserveTrade(
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: why did you change it? is linter ok with it?

File: contracts/KyberNetwork.sol [hunk 13]
```diff
@@ -392,54 +391,41 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
...
-                actualDestAmount,
-                KyberReserveInterface(rateResult.reserve2),
-                rateResult.rateEthToDest,
-                true));
+        require(doReserveTrade(ETH_TOKEN_ADDRESS, weiAmount, tradeInput.dest, tradeInput.destAddress, actualDestAmount,
+            KyberReserveInterface(rateResult.reserve2), rateResult.rateEthToDest, true));
 
         //when src is ether, reserve1 is doing a "fake" trade. (ether to ether) - don't burn.
         //when dest is ether, reserve2 is doing a "fake" trade. (ether to ether) - don't burn.
-        if (tradeInput.src != ETH_TOKEN_ADDRESS)
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: please undo these changes

File: contracts/KyberNetwork.sol [hunk 14]
```diff
@@ -469,8 +455,9 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface {
 
         if (src == dest) {
             //this is for a "fake" trade when both src and dest are ethers.
-            if (destAddress != (address(this)))
+            if (destAddress != (address(this))) {
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: I stopped reading here. we need to discuss general changes before we continue.

File: contracts/KyberNetwork.sol (hunk 6 above)
Comment (ilanDoron) [reply contracts/KyberNetwork.sol]: yes. added test.

**Comment by ilanDoron** [comment]
another PR was opened
